packer/cache.go

```diff
@@ -5,27 +5,29 @@ import (
 	"path/filepath"
 )
 
-var DefaultCacheDir = "packer_cache"
-
 // CachePath returns an absolute path to a cache file or directory
 //
-// When the directory is not absolute, CachePath will try to get
-// current working directory to be able to return a full path.
-// CachePath tries to create the resulting path if it doesn't exist.
-//
-// CachePath can error in case it cannot find the cwd.
+// When the directory is not absolute, CachePath will try to make a
+// a cache depending on the operating system.
 //
-// ex:
+// TODO: Update this with better examples
+// NOTE: cache directory will change depending on operating system dependent
+// For Windows:
 //   PACKER_CACHE_DIR=""            CacheDir() => "./packer_cache/
 //   PACKER_CACHE_DIR=""            CacheDir("foo") => "./packer_cache/foo
 //   PACKER_CACHE_DIR="bar"         CacheDir("foo") => "./bar/foo
 //   PACKER_CACHE_DIR="/home/there" CacheDir("foo", "bar") => "/home/there/foo/bar
+// For Unix:
+//   PACKER_CACHE_DIR="",            XDG_CONFIG_HOME="", Default_config CacheDir() => "$HOME/cache/packer"
+//   PACKER_CACHE_DIR="",            XDG_CONFIG_HOME="", Default_config CacheDir("foo") => "$HOME/cache/packer/foo"
+//   PACKER_CACHE_DIR="bar",         XDG_CONFIG_HOME="", Default_config CacheDir("foo") => "./bar/foo
+//   PACKER_CACHE_DIR="/home/there", XDG_CONFIG_HOME="", Default_config CacheDir("foo", "bar") => "/home/there/foo/bar
```

Comment:
I like these new examples !
What will `Default_config` stand for here ?
Do you think we should add something for `XDG_CONFIG_HOME` too ?
```suggestion
// For Unix:
//   PACKER_CACHE_DIR="",            XDG_CONFIG_HOME="", Default_config CacheDir() => "$HOME/cache/packer"
//   PACKER_CACHE_DIR="",            XDG_CONFIG_HOME="", Default_config CacheDir("foo") => "$HOME/cache/packer/foo"
//   PACKER_CACHE_DIR="bar",         XDG_CONFIG_HOME="", Default_config CacheDir("foo") => "./bar/foo
//   PACKER_CACHE_DIR="/home/there", XDG_CONFIG_HOME="", Default_config CacheDir("foo", "bar") => "/home/there/foo/bar
//   PACKER_CACHE_DIR="", XDG_CONFIG_HOME="/home/there", Default_config CacheDir("foo", "bar") => "/home/there/foo/bar
```